infrasim/model/elements/drive.py

```diff
@@ -79,6 +80,15 @@ def precheck(self):
             raise ArgsNotCorrect("[CBaseDrive] sector_size {} is only supported 'raw' format"
                                  .format(self.__sector_size))
 
+        if self._drive_info.get("file") and self._drive_info.get("size"):
```

Comment:
should check existence of file before querying its size. otherwise it can't get correct output in json, raises exception when getting value from json.